### PR TITLE
Upgrade to cert-manager 1.7

### PR DESCRIPTION
**Pull request by wallrj**
* Update to cert-manager 1.7
* Update to controller-runtime 0.11
* Update to compatible k8s dependencies 
* Build using Go 1.17
* Upgraded Kind to 0.12.0
* Try to get the Makefile working on MacOS to avoid commands and flags that aren't available there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -192,14 +192,6 @@ So it is important to add all the API types that are used in your issuer.
 
 Finally run `make generate manifests` again to update all the generated code.
 
-NOTE: You may encounter a dependency conflict between the version of controller-runtime used by cert-manager and the version installed by Kubebuilder.
-We have to use the cert-manager version and that in turn requires a newer version of the `zapr` logging library.
-Add the following line to `go.mod`:
-
-```text
-github.com/go-logr/zapr v0.2.0 // indirect
-```
```
Comment (wallrj) [inline README.md:201]: This comment is obsolete. There is no longer a conflict.

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
 // +build !ignore_autogenerated
```
Comment (wallrj) [inline api/v1alpha1/zz_generated.deepcopy.go:2]: This was changed by `go1.17 fmt`

File: internal/controllers/certificaterequest_controller_test.go [hunk 3]
```diff
@@ -597,7 +597,7 @@ func TestCertificateRequestReconcile(t *testing.T) {
 				Clock:                    fixedClock,
 			}
 			result, err := controller.Reconcile(
-				ctrl.LoggerInto(context.TODO(), &logrtesting.TestLogger{T: t}),
+				ctrl.LoggerInto(context.TODO(), logrtesting.NewTestLogger(t)),
```
Comment (wallrj) [inline internal/controllers/certificaterequest_controller_test.go:600]: They've exported a factory function and made TestLogger private.

**Comment by wallrj** [comment]
The `e2e` tests are not run automatically (yet) so I ran them locally:

```sh
$ make kind-cluster deploy-cert-manager docker-build kind-load deploy
...
```

```sh
$ make e2e
kubectl apply --filename config/samples
kubectl wait --for=condition=Ready --timeout=5s issuers.sample-issuer.example.com issuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificaterequests.cert-manager.io issuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificates.cert-manager.io certificate-by-issuer
kubectl wait --for=condition=Ready --timeout=5s clusterissuers.sample-issuer.example.com clusterissuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificaterequests.cert-manager.io clusterissuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificates.cert-manager.io certificate-by-clusterissuer
kubectl delete --filename config/samples
certificate.cert-manager.io/certificate-by-clusterissuer created
certificate.cert-manager.io/certificate-by-issuer created
certificaterequest.cert-manager.io/clusterissuer-sample created
certificaterequest.cert-manager.io/issuer-sample created
clusterissuer.sample-issuer.example.com/clusterissuer-sample unchanged
issuer.sample-issuer.example.com/issuer-sample unchanged
secret/clusterissuer-sample-credentials unchanged
secret/issuer-sample-credentials unchanged
issuer.sample-issuer.example.com/issuer-sample condition met
certificaterequest.cert-manager.io/issuer-sample condition met
certificate.cert-manager.io/certificate-by-issuer condition met
clusterissuer.sample-issuer.example.com/clusterissuer-sample condition met
certificaterequest.cert-manager.io/clusterissuer-sample condition met
certificate.cert-manager.io/certificate-by-clusterissuer condition met
certificate.cert-manager.io "certificate-by-clusterissuer" deleted
certificate.cert-manager.io "certificate-by-issuer" deleted
certificaterequest.cert-manager.io "clusterissuer-sample" deleted
certificaterequest.cert-manager.io "issuer-sample" deleted
clusterissuer.sample-issuer.example.com "clusterissuer-sample" deleted
issuer.sample-issuer.example.com "issuer-sample" deleted
secret "clusterissuer-sample-credentials" deleted
secret "issuer-sample-credentials" deleted

```

**Comment by wallrj** [comment]
I updated this PR to use the latest released cert-manager 1.7.0.

**Comment by wallrj** [comment]
@jahrlin I pushed some of those changes we just discussed to hopefully get the Makefile working on Mac M1 + darwin.

File: Makefile [hunk 4]
```diff
@@ -23,12 +23,12 @@ OS := $(shell go env GOOS)
...
 
 # Kind
-KIND_VERSION := 0.9.0
+KIND_VERSION := 0.12.0
 KIND := ${BIN}/kind-${KIND_VERSION}
 K8S_CLUSTER_NAME := sample-external-issuer-e2e
 
 # cert-manager
-CERT_MANAGER_VERSION ?= 1.3.0
+CERT_MANAGER_VERSION ?= 1.7.1
```
Comment (SgtCoDFish) [inline Makefile:31]: drive-by: 1.7.2 was released yday, might be worth using that instead :ok_hand:

**Review by jahrlin — APPROVED** [review]
Thanks @wallrj !

Works on my machine :)

File: Makefile (hunk 4 above)
Comment (wallrj) [reply Makefile:31]: I'll upgrade to the 1.8 alpha when it's ready, in a separate PR.
